### DISCUSSION TITLE
Two (2) db calls in .server

### DISCUSSION THREAD

**Discussion by ghost** [sveltejs/kit#6703] [General]
I have an edge case where I would like to run two separate db queries in .server and return both, in my case using SupaBase.

How can I setup 
const { data,count,error }=await supabase
...
const { data2,count2,error2 }=await supabase
...

To await both calls before return

return{
data1,
data2,
}

**Comment by LorisSigrist** [comment]
Take a look at [`Promise.allSettled()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) and [`Promise.all()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all).

**Reply by ghost** [reply]
Thank you. How would I wrap Promise.all correctly around above example?

**Reply by teemingc** [reply]
> Thank you. How would I wrap Promise.all correctly around above example?

Perhaps something like this
```ts
const query1 = supabase
const query2 = supabase
const [res1, res2] = await Promise.all([query1, query2]);

return {
  data1: res1.data,
  data2: res2.data
}
```

**Reply by ghost** [reply]
Thank you

**Comment by ZetiMente** [comment]
why can't u just do ?

const query1 = await supabase...
const query2 = await supabase...
		
return {
query1,
query2
};

**Reply by ghost** [reply]
Thanks. That way query2 wont be picked up before the return is issued.

**Comment by bjon** [comment]
If i'm not mistaken, you can return promises directly. No need to use await.
